crates/core/src/client/mod.rs

```diff
@@ -757,4 +777,112 @@ impl<P: Provider + Send + Sync> KakarotStarknetApi<P> for KakarotClient<P> {
 
         Ok(events)
     }
+
+    async fn check_eoa_account_exists(
+        &self,
+        ethereum_address: Address,
+        starknet_block_id: &StarknetBlockId,
+    ) -> Result<bool, EthApiError<P::Error>> {
+        let eoa_account_starknet_address = self.compute_starknet_address(ethereum_address, starknet_block_id).await?;
+
+        let result = self
+            .starknet_provider
+            .call(
+                FunctionCall {
+                    contract_address: eoa_account_starknet_address,
+                    entry_point_selector: GET_EVM_ADDRESS,
+                    calldata: vec![],
+                },
+                starknet_block_id,
+            )
+            .await;
+
+        let result: Result<bool, ProviderError<<P as Provider>::Error>> = match result {
+            Ok(_) => Ok(true),
+            Err(error) => match error {
+                ProviderError::StarknetError(error) => match error {
+                    StarknetError::ContractNotFound => Ok(false),
+                    _ => Err(ProviderError::StarknetError(error)),
+                },
+                _ => Err(error),
+            },
+        };
+
+        Ok(result?)
+    }
+
+    async fn deploy_eoa(&self, ethereum_address: Address) -> Result<FieldElement, EthApiError<P::Error>> {
+        let account = &self.starknet_account;
+        let kakarot_address = self.kakarot_address();
+
+        let ethereum_address: FieldElement = Felt252Wrapper::from(ethereum_address).into();
+
+        let result = account
+            .execute(vec![Call {
+                calldata: vec![ethereum_address],
+                to: kakarot_address,
+                selector: DEPLOY_EXTERNALLY_OWNED_ACCOUNT,
+            }])
+            .send()
+            .await;
+
+        let result: Result<FieldElement, EthApiError<P::Error>> = match result {
+            Ok(invoke_result) => Ok(invoke_result.transaction_hash),
+            Err(error) => match error {
+                AccountError::Provider(error) => Err(EthApiError::from(error)),
+                _ => Err(EthApiError::Other(error.into())),
+            },
+        };
+
+        result
```

Comment:
We have a kakarot contract abstraction, can you please add a `deploy_externally_owned_account` method to it and move this code there?